torch/lowrank.py

```diff
@@ -0,0 +1,325 @@
+"""Implement various linear algebra algorithms for low rank matrices.
+"""
+
+import torch
+
+
+def is_sparse(A):
+    """Check if tensor A is a sparse tensor"""
+    if isinstance(A, torch.Tensor):
+        return A.layout == torch.sparse_coo
+    import scipy.sparse
```

Comment:
same about import

Reply:
scipy and numpy imports are now eliminated.